packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -370,20 +370,23 @@ def async_act(self) -> Generator:
 
     def has_transaction_been_sent(self) -> Generator[None, None, Optional[bool]]:
         """Transaction verification."""
+
+        if not self.synchronized_data.tx_hashes_history:
+            return False
+
+        to_be_validated_tx_hash = self.synchronized_data.tx_hashes_history[-1]
```

Comment:
Why not use the `to_be_validated_tx_hash` property instead and check for `None` to return early?